### DISCUSSION TITLE
Version 1.1 Build 2

### DISCUSSION THREAD

**Discussion by ErinaSugino** [ErinaSugino/Starbound-SBR-Mod#51] [Update Builds]
## Version 1.1 Patchnotes
- Integrated Lustbound into SBR
- Added modular pregnancy management (Including multi-baby pregnancies)
- Added new cumflation and fertility stacking mechanics (Thanks to XspeedPL)
- Added properly working baby genetics
- Re-added global NPC auto-climax
- Outsourced UI sync data for easy patching
- Added stillborn chance
- Added True Kid NPCs
- Fixed incompatibility with "Npc Drops Gear" mod

## Build 2 Notes
- Build is not tested. This pre-release is for public testing purposes only.
- Build 2's testing focus should be set upon ensuring that True Kid NPCs are fully safe from sexual content.

### Important Notes
- Do not use on main savefile - 1.1 changes the data format for pregnancies which is not backwards compatible. Reverting back to 1.0 will erase or break all ongoing pregnancies!

<hr /><em>This discussion was created from the release <a href='https://github.com/ErinaSugino/Starbound-SBR-Mod/releases/tag/v1.1_b2'>Version 1.1 Build 2</a>.</em>

**Comment by 777joshua7771** [comment]
Righty, got another Syntax error after a bit from shooting a npc with the aphrodites bow, resulting in the npc's death.
Alongside that, there seem to be another error, not sure if it really a Syntax but likely pregnancy related, it's seem to failed to spawn in a new npc after naming them.
Both errors are in the same log, Syntax error is at line 679, while the naming new NPC (Birthing) is at line 1568.

[starbound.log](https://github.com/ErinaSugino/Starbound-SBR-Mod/files/12717277/starbound.log)

**Reply by 777joshua7771** [reply]
Not really sure if i should've made a new issue to have it posted there, which i can do if required.

**Reply by ErinaSugino** [reply]
There you go.

**Reply by 777joshua7771** [reply]
Nice, and i think that might be the last errors fixed, got none from the mod itself after testing it out.

**Reply by ErinaSugino** [reply]
But the important question is - what happens after you successfully gave birth?

**Reply by 777joshua7771** [reply]
No errors seem to pop up as it ended up successful with the birth, alongside with the small avian spawning in place.

![starbound_SgflSHVgBM](https://github.com/ErinaSugino/Starbound-SBR-Mod/assets/93668917/79d0b99e-4294-4ac4-a87d-1600a4e42e51)

**Reply by ErinaSugino** [reply]
Huh. It just working:tm: is a new one. xD

Does the NPC retain it's name if you give it one?
Does it somewhat resemble (in colors) the parents, or is it a broken "default color" NPC?
Does it work with funky custom races too?

**Reply by 777joshua7771** [reply]
Though, if there were any special form of animation that meant to play out, which i assume there isn't at the time being, it didn't seem to play.

**Reply by 777joshua7771** [reply]
Oh, haven't seen that post, but think there was some things that you mentioned now:
Both my character and the father are around blue-purple, so it might've still ended up default color npcs.

**Reply by ErinaSugino** [reply]
Gimme the log.

Also, in terms of animation the player is supposed to be forced into a solo sexnode playing a one-frame "animation" of lying down for now. If the spawning of the node fails it should be in the log.

**Reply by 777joshua7771** [reply]
I'm about to check out if the name retains, and i may try it as the avali race soon.

**Reply by 777joshua7771** [reply]
Alrighty, also, there **are** a few error info, but most i had checked so far is either another mod or male avian missing body sprite.

**Reply by 777joshua7771** [reply]
[starbound.log](https://github.com/ErinaSugino/Starbound-SBR-Mod/files/12717609/starbound.log)

Not to alarm about missing male avian sprite, it reverts to it's placeholder default.

**Reply by 777joshua7771** [reply]
Oh right, maybe i should start turning off info from the sexbound config unless it have any important info set to it, i've noticed it's been clustering up the logs.

**Reply by 777joshua7771** [reply]
To confirm the name situlation, Newly spawned NPC retains the name you gave it, both random or typed.
And it does work with custom race if it same-race intercourse.

**Reply by ErinaSugino** [reply]
Logs for testing, troubleshooting and debugging always need all levels of logging enabled.

So, what this log tells me is that the player was successfully stuffed into a sexnode during the birthing process (which you probably don't see much of because of the birthing UI popups).
Which created a baby named "Cipilli" - which sounds pretty much auto generated. Also, despite there being a warning about "invalid pregnancy, aborting"... it still continued and succeeded? There' something wrong here.
Also, it detected that the closest color paletted of the parents are number 2 and number 7 from the species file.

Can you do "/entityeval npc.humanoidIdentity()" on both the father NPC and the born child NPC and give me the results from your log?

**Reply by 777joshua7771** [reply]
Alrighty then, and i don't tend to usually give names to NPC or sometime my character cause of not being creative with thinking up of a name, but with the avali testing before, i decided to give a second avali "abc" for a name, and it did kept the name.
I'll go back onto my avian character to check out both the father and child NPCs.

**Reply by 777joshua7771** [reply]
Back again with the log containing the information of both the NPCs from my testing avian character.
[starbound.log](https://github.com/ErinaSugino/Starbound-SBR-Mod/files/12717773/starbound.log)

**Reply by ErinaSugino** [reply]
Ok, the generated colors don't actually get applied. Great.

**Reply by 777joshua7771** [reply]
Hopfully that's be a quick fix, same goes with the animation for the birthing part.

**Reply by ErinaSugino** [reply]
Haven't yet even reached the part about colors, but there is a lot wrong here. A wonder it even "works" in the first place.

**Reply by 777joshua7771** [reply]
Huh, maybe lady luck had spared us once from further problems that prevents the whole pregnancy system to work entirely.
Overall, having quite a feeling that one of the codes that's in the wrong might cause problems after fixing, but it's a guess on my end.

**Reply by ErinaSugino** [reply]
New zip fixes the biggest problems/inconsistencies encountered during that birth test. No update on colors yet though.

Have to look into statistics and reverse cowgirl.

**Reply by ErinaSugino** [reply]
Cannot confirm any issues with reverse cowgirl on a table, at least not with a human and a Neki and after I removed Relayer.
